### PR TITLE
#74845 Retain pipeline run

### PR DESCRIPTION
**Pull request by ru-ukho**

### CODE REVIEW

**Comment by srjames-ukho** [comment]
Hi Ravi, can you update the main README.md in the root directory to the direction to the README.md you added. Cheers

File: retain-pipeline/README.md
Comment (srjames-ukho) [inline retain-pipeline/README.md:1]: Hi Ravi, can you update the main README.md in the root directory to the direction to the README.md you added. Cheers

File: retain-pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      endpoint: GitHub_Service_Connection_Name
+      name: UKHO/devops-pipelinetemplates
+```
+
+2. Add a job to your pipeline with the required parameters to create retention lease.
+   Below is an example 
+
+
+```yaml
+- job: githubtag
```
Comment (srjames-ukho) [inline retain-pipeline/README.md:24]:
```suggestion
- job:
```

File: retain-pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     - checkout: UKHOTemplates
+     - template: retain-pipeline/retain-pipeline.yml@UKHOTemplates
+             parameters:
+                DaysValid: 365
+                AccessToken: "$(System.AccessToken)"
+                DefinitionId: '$(System.DefinitionId)'
+                OwnerId: '$(Build.RequestedForId)'
+                BuildId: '$(Build.BuildId)'
+                CollectionUri: '$(System.CollectionUri)'
+                TeamProject: '$(System.TeamProject)'
```
Comment (srjames-ukho) [inline retain-pipeline/README.md:37]:
```suggestion
        dependsOn: "<DEPENDS ON JOB>"
        condition: "succeeded('<CONDITIONAL JOB>')"
        steps:  
           - checkout: UKHOTemplates
           - template: retain-pipeline/retain-pipeline.yml@UKHOTemplates
             parameters:
                DaysValid: 365
                AccessToken: "$(System.AccessToken)"
                DefinitionId: '$(System.DefinitionId)'
                OwnerId: '$(Build.RequestedForId)'
                BuildId: '$(Build.BuildId)'
                CollectionUri: '$(System.CollectionUri)'
                TeamProject: '$(System.TeamProject)'
```

File: retain-pipeline/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - repository: UKHOTemplates
+      type: github
+      endpoint: GitHub_Service_Connection_Name
+      name: UKHO/devops-pipelinetemplates
+```
+
+2. Add a job to your pipeline with the required parameters to create retention lease.
+   Below is an example 
+
+
```
Comment (srjames-ukho) [inline retain-pipeline/README.md:22]:
```suggestion
```

**Review by srjames-ukho — CHANGES_REQUESTED** [review]
Mostly some formatting in the example yaml. Could do with the root README.md updating with a reference to your new README.md file.

**Review by srjames-ukho — COMMENTED** [review]
Dismissing my request changes status as I will be annual leave next week and don't want that status to block the PR in my absense.

**Review by srjames-ukho — APPROVED** [review]

**Review by srjames-ukho — COMMENTED** [review]
Dismissed my request change status, should be safe for next week now

File: retain-pipelinerun/retain-pipelinerun.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  - checkout: self
+    persistCredentials: true
+
+  - task: PowerShell@2
+    condition: and(succeeded(), not(canceled()))
+    displayName: Retain Live Release
+    inputs:
+      targetType: filePath
+      filePath: ./devops-pipelinetemplates/retain-pipelinerun/add-lease.ps1
+      arguments: "-daysValid ${{parameters.DaysValid}}
```
Comment (elduddz) [inline retain-pipelinerun/retain-pipelinerun.yml:17]: please change how these are provided:
https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/powershell-v2?view=azure-pipelines#call-powershell-script-with-multiple-arguments

File: retain-pipelinerun/retain-pipelinerun.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+- name: DaysValid
+  type: number
+  default: 30
+
+
+steps:
+  - checkout: self
+    persistCredentials: true
+
+  - task: PowerShell@2
```
Comment (elduddz) [inline retain-pipelinerun/retain-pipelinerun.yml:11]: please add pwsh: true to help with use of powershell core.

**Review by elduddz — CHANGES_REQUESTED** [review]

**Review by elduddz — APPROVED** [review]

**Pull request merged**